app/assets/javascripts/Components/Result/result.jsx

```diff
@@ -33,6 +33,7 @@ const INITIAL_FILTER_MODAL_STATE = {
     min: "",
     max: "",
   },
+  criteria: [],
```

Comment:
This is my most substantive comment: overall for the data model, this state should be an object that maps criterion name to `{min: ..., max: ...}` rather than a list. This change facilitates more efficient lookup by criterion name, which is the main functionality you need for updating this state.